packages/arb-provider-ethers/src/lib/abi/index.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  BigNumberish,
+  EventDescription,
+  FunctionDescription,
+} from 'ethers/utils'
+
+export class TransactionOverrides {
+  nonce?: BigNumberish | Promise<BigNumberish>
+  gasLimit?: BigNumberish | Promise<BigNumberish>
+  gasPrice?: BigNumberish | Promise<BigNumberish>
+  value?: BigNumberish | Promise<BigNumberish>
+  chainId?: number | Promise<number>
+}
+
+export interface TypedEventDescription<
+  T extends Pick<EventDescription, 'encodeTopics'>
+> extends EventDescription {
+  encodeTopics: T['encodeTopics']
+}
+
+export interface TypedFunctionDescription<
+  T extends Pick<FunctionDescription, 'encode'>
+> extends FunctionDescription {
+  encode: T['encode']
+}
```

Comment:
forgot to explain this. I copied this from the typechain generated `index.d.ts` file. Since that file is generated, it can be overwritten at anytime. 

When building this project with `tsc`, the `index.ts` file takes precedence over the `.d.ts` file, so the types aren't copied into the build folder (not that they would be copied over without the presence of `index.ts` - typescript doesn't emit anything for declaration only files).

this is the simplest way of having a directory index while also maintaining these types (which are depended on by the rest of the package)

Reply:
Ah, I missed this going through. That makes sense and seems like a good approach.